src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -1,153 +1,163 @@
-                .map(Section::getUpStation)
-                .collect(Collectors.toList());
-        List<Station> downStations = this.sections.stream()
-                .map(Section::getDownStation)
-                .collect(Collectors.toList());
-
-        return upStations.stream()
-                .filter(it -> !downStations.contains(it))
-                .findFirst()
-                .orElseThrow(RuntimeException::new);
-    }
-
-    private void addNewSectionForDelete(Optional<Section> upSection, Optional<Section> downSection) {
-        if (upSection.isPresent() && downSection.isPresent()) {
-            Section newSection = new Section(
-                    upSection.get().getLine(),
-                    downSection.get().getUpStation(),
-                    upSection.get().getDownStation(),
-                    upSection.get().getDistance() + downSection.get().getDistance()
-            );
-
-            this.sections.add(newSection);
-        }
-    }
-
-    private Optional<Section> findSectionAsUpStation(Station finalUpStation) {
-        return this.sections.stream()
-                .filter(it -> it.isSameUpStation(finalUpStation))
-                .findFirst();
-    }
-
-    private Optional<Section> findSectionAsDownStation(Station station) {
-        return this.sections.stream()
-                .filter(it -> it.isSameDownStation(station))
-                .findFirst();
-    }
-
-    public int totalDistance() {
-        return sections.stream().mapToInt(Section::getDistance).sum();
-    }
+	@OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+	private List<Section> sections = new ArrayList<>();
+
+	public Sections() {
+	}
+
+	public Sections(List<Section> sections) {
+		this.sections = sections;
+	}
+
+	public List<Section> getSections() {
+		return sections;
+	}
+
+	public void add(Section section) {
+		if (this.sections.isEmpty()) {
+			this.sections.add(section);
+			return;
+		}
+
+		checkDuplicateSection(section);
+
+		rearrangeSectionWithUpStation(section);
+		rearrangeSectionWithDownStation(section);
+
+		sections.add(section);
+	}
+
+	public void delete(Station station) {
+		if (this.sections.size() <= 1) {
+			throw new IllegalArgumentException();
+		}
+
+		Optional<Section> upSection = findSectionAsUpStation(station);
+		Optional<Section> downSection = findSectionAsDownStation(station);
+
+		addNewSectionForDelete(upSection, downSection);
+
+		upSection.ifPresent(it -> this.sections.remove(it));
+		downSection.ifPresent(it -> this.sections.remove(it));
+	}
+
+	public List<Station> getStations() {
+		if (this.sections.isEmpty()) {
+			return Collections.emptyList();
+		}
+
+		Station upStation = findFirstUpStation();
+		List<Station> result = new ArrayList<>();
+		result.add(upStation);
+
+		while (true) {
+			Station finalUpStation = upStation;
+			Optional<Section> section = findSectionAsUpStation(finalUpStation);
+
+			if (!section.isPresent()) {
```

Comment:
optional도 자바 11부터 isPresent반대인 isEmpty 로 사용 가능합니다 👍 
```suggestion
			if (section.isEmpty()) 
```
[참고 내용](https://daddyprogrammer.org/post/1988/java-optional/)